kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala

```diff
@@ -174,9 +174,11 @@ class KyuubiSessionImpl(
               } else {
                 Option(password).filter(_.nonEmpty).getOrElse("anonymous")
               }
-            _client = KyuubiSyncThriftClient.createClient(user, passwd, host, port, sessionConf)
+            val engineClient =
+              KyuubiSyncThriftClient.createClient(user, passwd, host, port, sessionConf)
             _engineSessionHandle =
-              _client.openSession(protocol, user, passwd, openEngineSessionConf)
+              engineClient.openSession(protocol, user, passwd, openEngineSessionConf)
+            _client = engineClient
```

Comment:
looks reasonable, and `_client.closeSession()` in finally cause is useless then

Reply:
let me check

Reply:
how about now? @pan3793

Reply:
thanks for updating, looks good now.